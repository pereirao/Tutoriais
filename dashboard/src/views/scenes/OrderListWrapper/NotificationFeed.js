// @flow

import React from 'react';
import {connect} from 'react-redux';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import * as Ent from '@minibar/store-business/src/utils/ent';
import {notification_selectors} from '../../../business/notification';
import style_sheet from './OrderListWrapper.module.css';
import NotificationCard from './NotificationCard';

// TODO: on scroll virtualize
type NotificationFeedProps = { notifications: Array<Notification> };
export const NotificationFeed = ({notifications}: NotificationFeedProps) => {
  return (
    <ReactCSSTransitionGroup
      className={style_sheet.feedWrapper}
      transitionAppear
      transitionAppearTimeout={700}
      transitionName={style_sheet}
      transitionEnterTimeout={700}
      transitionLeaveTimeout={700}>

      {notifications.map(notification => (
        <NotificationCard
          key={notification.id}
          notification_id={notification.id}
          notification_type={notification.notification_type}
          order_id={notification.order_id}/>
      ))}
    </ReactCSSTransitionGroup>
  );
};

const NotificationFeedSTP = () => {
  const finder = Ent.find('notification');
  return state => ({
    notifications: finder(state, notification_selectors.displayOrder(state))
  });
};
const NotificationFeedContainer = connect(NotificationFeedSTP)(NotificationFeed);
export default NotificationFeedContainer;
