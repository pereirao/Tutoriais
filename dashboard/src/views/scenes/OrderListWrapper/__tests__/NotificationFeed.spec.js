import React from 'react';

import notification_factory from '../../../../business/notification/__tests__/notification.factory';
// NOTE: the named import is not connected to the redux store
import { NotificationFeed } from '../NotificationFeed';

it('renders feed correctly', () => {
  expect(shallow(<NotificationFeed notifications={[notification_factory.build({id: 1}), notification_factory.build({id: 2})]} />)).toMatchSnapshot();
});
