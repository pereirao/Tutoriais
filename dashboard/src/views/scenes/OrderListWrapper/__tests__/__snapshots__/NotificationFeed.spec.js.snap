// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders feed correctly 1`] = `
<CSSTransitionGroup
  className="feedWrapper"
  transitionAppear={true}
  transitionAppearTimeout={700}
  transitionEnter={true}
  transitionEnterTimeout={700}
  transitionLeave={true}
  transitionLeaveTimeout={700}
  transitionName={Object {}}
>
  <Connect(NotificationCard)
    notification_id={1}
    notification_type="comment"
    order_id={10}
  />
  <Connect(NotificationCard)
    notification_id={2}
    notification_type="comment"
    order_id={10}
  />
</CSSTransitionGroup>
`;
